### ISSUE TITLE
Many stocks disabled - jaggedy recoil

### ISSUE THREAD

**Issue by CodeGenStudio** [Mr-Pineapple/AdditionalGuns#63]
So I'm new to using this mod and getting used to it, I've been pretty satisfied that it's basically in-differentiable (in terms of attachments, crafting) from the basic Gun mod by Mr-CrayFish and all the handguns are very impressive.

Just something I noticed that was pretty annoying, is that most of the AK-series guns, and M4Ax guns as well, is that the recoil upwards is way too much, and too jaggedy - meaning, it jerks violently up and makes it extremely difficult to aim and handle and not so eye-friendly either. What makes this worse is the fact i can't add any custom stock so I'm stuck with default butt and the silencer only does so much.

**Comment by CasualNoLifer** [comment]
> So I'm new to using this mod and getting used to it, I've been pretty satisfied that it's basically in-differentiable (in terms of attachments, crafting) from the basic Gun mod by Mr-CrayFish and all the handguns are very impressive.
> 
> Just something I noticed that was pretty annoying, is that most of the AK-series guns, and M4Ax guns as well, is that the recoil upwards is way too much, and too jaggedy - meaning, it jerks violently up and makes it extremely difficult to aim and handle and not so eye-friendly either. What makes this worse is the fact i can't add any custom stock so I'm stuck with default butt and the silencer only does so much.

Are you using the 0.7.0 version of the mod? Also, which guns are you referring to specifically? I recently headed a patch that introduced an overall increase to recoil in all guns but they were all buffed. If you're talking about specifically the AKM/AK74, they are intended to be more of a challenge weapon as they have relatively high recoil and no attachment slots.

Having guns in Minecraft is very overpowered and difficult to balance for PVP and gameplay -- more difficult ammo crafting and higher recoil is the way that we chose to make the guns harder to maintain as weapons.

**Comment by CodeGenStudio** [comment]
> > So I'm new to using this mod and getting used to it, I've been pretty satisfied that it's basically in-differentiable (in terms of attachments, crafting) from the basic Gun mod by Mr-CrayFish and all the handguns are very impressive.
> > Just something I noticed that was pretty annoying, is that most of the AK-series guns, and M4Ax guns as well, is that the recoil upwards is way too much, and too jaggedy - meaning, it jerks violently up and makes it extremely difficult to aim and handle and not so eye-friendly either. What makes this worse is the fact i can't add any custom stock so I'm stuck with default butt and the silencer only does so much.
> 
> Are you using the 0.7.0 version of the mod? Also, which guns are you referring to specifically? I recently headed a patch that introduced an overall increase to recoil in all guns but they were all buffed. If you're talking about specifically the AKM/AK74, they are intended to be more of a challenge weapon as they have relatively high recoil and no attachment slots.
> 
> Having guns in Minecraft is very overpowered and difficult to balance for PVP and gameplay -- more difficult ammo crafting and higher recoil is the way that we chose to make the guns harder to maintain as weapons.

Yes I'm using version 0.7.0 on Minecraft 1.16.5. Your reasoning makes sense and I pretty much agree. But would it be possible to make that high recoil a little less 'brutal'? Like the animation to be smooth going up, instead of a violent jerk-up so it's easier on the eyes. 

> If you're talking about specifically the AKM/AK74

Yesss, these are the main ones I noticed them on.

**Comment by CasualNoLifer** [comment]
> > > So I'm new to using this mod and getting used to it, I've been pretty satisfied that it's basically in-differentiable (in terms of attachments, crafting) from the basic Gun mod by Mr-CrayFish and all the handguns are very impressive.
> > > Just something I noticed that was pretty annoying, is that most of the AK-series guns, and M4Ax guns as well, is that the recoil upwards is way too much, and too jaggedy - meaning, it jerks violently up and makes it extremely difficult to aim and handle and not so eye-friendly either. What makes this worse is the fact i can't add any custom stock so I'm stuck with default butt and the silencer only does so much.
> > 
> > 
> > Are you using the 0.7.0 version of the mod? Also, which guns are you referring to specifically? I recently headed a patch that introduced an overall increase to recoil in all guns but they were all buffed. If you're talking about specifically the AKM/AK74, they are intended to be more of a challenge weapon as they have relatively high recoil and no attachment slots.
> > Having guns in Minecraft is very overpowered and difficult to balance for PVP and gameplay -- more difficult ammo crafting and higher recoil is the way that we chose to make the guns harder to maintain as weapons.
> 
> Yes I'm using version 0.7.0 on Minecraft 1.16.5. Your reasoning makes sense and I pretty much agree. But would it be possible to make that high recoil a little less 'brutal'? Like the animation to be smooth going up, instead of a violent jerk-up so it's easier on the eyes.
> 
> > If you're talking about specifically the AKM/AK74
> 
> Yesss, these are the main ones I noticed them on.

I will definitely change the recoil on those two since I did notice they're a bit rough to use with the high recoil. I would suggest using the modular versions of them (AKM Custom and AK74m) to help with recoil. I did make the recoil the way it is to be a little more realistic. If we get more complaints about the AK recoil being visually unappealing I can revert the change or nerf the recoil some.

**Comment by CodeGenStudio** [comment]
> I did make the recoil the way it is to be a little more realistic. If we get more complaints about the AK recoil being visually unappealing I can revert the change or nerf the recoil some.

Fair enough, thanks a bunch and good work on the mod mate :))

**Comment by CasualNoLifer** [comment]
> > I did make the recoil the way it is to be a little more realistic. If we get more complaints about the AK recoil being visually unappealing I can revert the change or nerf the recoil some.
> 
> Fair enough, thanks a bunch and good work on the mod mate :))

Thanks! Also thank Autovw if you ever see him. I just joined the team for the recent update! haha.
Feel free to ask any other questions if you need them. Enjoy!
